src/data/transforms/mod.rs

```diff
@@ -21,6 +21,8 @@ pub use self::standardize::Standardizer;
 
 /// Trait for data transformers
 pub trait Transformer<T> {
+    /// Fit Transformer to input data, and stores the transformation in the Transformer
+    fn fit(&mut self, inputs: &T) -> Result<(), error::Error>;
     /// Transforms the inputs and stores the transformation in the Transformer
```

Comment:
I would add to this comment and state that if this function is used without `fit`ting first then the `Transformer` will call `fit` itself.